cfme/tests/physical_infrastructure/ui/test_physical_server_list_buttons.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+import collections
+import pytest
+
+from cfme.common.physical_server_views import PhysicalServersView
+from cfme.physical.provider.lenovo import LenovoProvider
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.wait import wait_for
+
+pytestmark = [pytest.mark.tier(3), pytest.mark.provider([LenovoProvider], scope="module")]
+
+
+@pytest.fixture(scope="function")
+def physical_servers(appliance, provider, setup_provider):
+    # Get and return all the physical servers
+    return appliance.collections.physical_servers.all(provider)
+
+
+@pytest.fixture(scope="function")
+def physical_servers_collection(appliance):
+    # Get and return collection of the physical servers
+    return appliance.collections.physical_servers
+
+
+# Configuration Button
+def test_refresh_relationships(physical_servers_collection, physical_servers, provider):
+    view = navigate_to(physical_servers_collection, "All")
+    last_refresh = provider.last_refresh_date()
+    item = "Refresh Relationships and Power States"
+    physical_servers_collection.custom_button_action("Configuration", item, physical_servers)
+    out, time = wait_for(
+        lambda: last_refresh != provider.last_refresh_date(),
+        fail_func=view.browser.refresh,
+        message="Wait for the servers to be refreshed...",
+        num_sec=300,
+        delay=5
+    )
+    assert out
+
+
+Action = collections.namedtuple('Action', 'button item method')
+actions = [
+    Action("Power", "Power Off", "power_off"),
+    Action("Power", "Power On", "power_on"),
+    Action("Power", "Power Off Immediately", "power_off_now"),
+    Action("Power", "Restart", "restart"),
+    Action("Power", "Restart Immediately", "restart_now"),
+    Action("Power", "Restart to System Setup", "restart_to_sys_setup"),
+    Action("Power", "Restart Management Controller", "restart_mgmt_controller"),
+    Action("Identify", "Blink LED", "blink_loc_led"),
+    Action("Identify", "Turn Off LED", "turn_off_loc_led"),
+    Action("Identify", "Turn On LED", "turn_on_loc_led")
+]
+
+
+# Power / Identify Buttons
+@pytest.mark.parametrize("button, item, method",
+                         actions, ids=[action.item for action in actions])
+def test_server_actions(physical_servers_collection, physical_servers, provider,
+                        button, item, method):
+    """ Test the physical server actions are creating a handler alert to each action of the a collection
+    of physical servers.
+    Params:
+        * button: the button to be performed on the physical server list page
+        * item: the item to be selected inside the dropdrown button
+        * method: the name of the method that most be used to compare if was invoked the
+        current method on the manageIQ.
+    Metadata:
+        test_flag: crud
+    """
+    view = provider.create_view(PhysicalServersView)
```

Comment:
Test needs navigation - `custom_button_action` doesn't navigate.

Reply:
yap, but this method `select_entity_rows` inside custom_button_action navigate to current page. I'm running here and I'm using the VNC viewer to follow the selenium navigate on the views.

Reply:
@mshriver